### PR TITLE
Nginx docker improvements

### PR DESCRIPTION
**Pull request by kwypchlo**
- use docker-entrypoint script from official docker-nginx repo to improve task management
  - replace manual templating with envsubst entrypoint script from official docker-nginx repo
    - it doesn't support conditionals but I found out that even if we have 2 nginx servers with the same name in configuration, it will properly skip the latter which is what we actually did with the conditions
  - replace inline CMD script for reloading nginx config every 6 hours (for letsencrypt) with entrypoint script (copied from pull request on docker-nginx repo)
  - replace build step that produces local ssl certificate with entrypoint script that does exactly that - we probably don't want to include this in build step since it will end up in the image when we build it and deploy on dockerhub (probably bad practice)
- enable pcre_jit to speed up regex processing (copied from openresty nginx defaults)
- replaced copying over nginx configuration in dockerfile to mounting it with volumes
- use [skynetlabs/nginx](https://hub.docker.com/r/skynetlabs/nginx) docker image from dockerhub
- add docker/nginx/testing/Dockerfile to hadolint

### CODE REVIEW

File: docker/nginx/Dockerfile [hunk 1]
```diff
@@ -2,25 +2,20 @@ FROM openresty/openresty:1.19.9.1-focal
...
+    apt-get clean && rm -rf /var/lib/apt/lists/* && \
+    luarocks install lua-resty-http && \
+    luarocks install hasher
 
-COPY mo ./
-COPY libs /etc/nginx/libs
-COPY conf.d /etc/nginx/conf.d
-COPY conf.d.templates /etc/nginx/conf.d.templates
-COPY nginx.conf /usr/local/openresty/nginx/conf/nginx.conf
+# reload nginx every 6 hours (for reloading certificates)
```
Comment (peterjan) [inline docker/nginx/Dockerfile:10]: 🕺  woop woop - nice

**Review by peterjan — APPROVED** [review]

File: docker/nginx/conf.d.templates/server.hns.conf.template [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	
+	include /etc/nginx/conf.d/server/server.http;
+
+	set_by_lua_block $server_alias { return string.match("${SERVER_DOMAIN}", "^([^.]+)") }
+}
+
+server {
+	server_name *.hns.${SERVER_DOMAIN}; # example: *.hns.eu-ger-1.siasky.net
+	
+	set_by_lua_block $skynet_portal_domain { return "${SERVER_DOMAIN}" }
```
[the remark this answers is not in the document]
Comment (kwypchlo) [reply docker/nginx/conf.d.templates/server.hns.conf.template:34]: yeah, I'll add a comment

**Review by ro-tex — DISMISSED** [review]
LGTM.

Parts of the new code are quite hard to review, so I basically trust that they behave in the expected way and that this has been manually tested to give the right results.

**Comment by kwypchlo** [comment]
> LGTM.
> 
> Parts of the new code are quite hard to review, so I basically trust that they behave in the expected way and that this has been manually tested to give the right results.

I renamed the files and deleted some conditions (deployed and tested) that's why they appear new

**Review by peterjan — APPROVED** [review]

**Review by ro-tex — APPROVED** [review]

**Pull request merged**